### DISCUSSION TITLE
Repeated identifiers in Nexus resolver

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#2715] [Q&A] [answered]
Hi, 
is there a way not to write all fields 3x? I don't want to use t.crud as I still need the access to the function.
E.g. change this 
 ```ts
   t.field('createDraft', {
      type: 'Post',
      args: {
        title: stringArg({ nullable: false }),
        content: stringArg(),
        authorEmail: stringArg({ nullable: false }),
      },
      resolve: (_, { title, content, authorEmail }, ctx) => ctx.prisma.post.create({
        data: {
          title,
          content,
          published: false,
          author: {
            connect: { email: authorEmail },
          },
        },
      }),
    })

```
Into this
```ts
    t.field('createDraft', {
      type: 'Post',
      args: argDefs,
      resolve: (_, argVals, ctx) => ctx.prisma.post.create({
        data: argVals,
      }),
    })
```

Thanks :)

**Comment by ryands17** [comment, marked as answer]
Hey @TomasMoskala :wave: 
That is currently not possible in Nexus, so you would have to explicitly specify the fields as a workaround for now. There is an [open issue](https://github.com/graphql-nexus/nexus-schema-plugin-prisma/issues/303) as well for this use case.

**Reply by TomasMoskala** [reply]
Hi, not even this?

```ts
    t.field('login', {
      type: 'user',
      args: {
        user_email: stringArg({ nullable: false }), 
      },
      async resolve(_, args , ctx) { 
        const result = await ctx.prisma.user.findOne({ 
          where: { args.user_email },      
        });
        if (result === null) {
          throw new Error(`No email "${args.user_email}"`)
        }
        return result
      }
    })

```
Funny that where: { args.user_email } shows an error
Type '{ args: { user_email: string; }; "": any; }' is not assignable to type 'userWhereUniqueInput'.
but  "${args.user_email}" is OK even with autocompletion :)
![image](https://user-images.githubusercontent.com/35471478/84440735-cb78e580-ac3a-11ea-88b3-996ef1462117.png)
![image](https://user-images.githubusercontent.com/35471478/84440898-13980800-ac3b-11ea-9b9c-209c4a16f74a.png)

**Reply by ryands17** [reply]
I think it should be:

```ts
t.field('login', {
      type: 'user',
      args: {
        user_email: stringArg({ nullable: false }), 
      },
      async resolve(_, args , ctx) { 
        const result = await ctx.prisma.user.findOne({ 
          where: { user_email: args.user_email },      
        });
        if (result === null) {
          throw new Error(`No email "${args.user_email}"`)
        }
        return result
      }
    })
```

**Reply by TomasMoskala** [reply]
Correct, making the code even bigger when creating new rows, so I will stick to the duplicating. Thanks!

**Comment by ludralph** [comment]
Hi @TomasMoskala,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
